Tests/TestNonSimpleCases/FixtureNonSimpleFuncs.swift

```diff
@@ -514,3 +514,33 @@ class NonSimpleFuncsMock: NonSimpleFuncs {
     }
 }
 """
+
+let returnSelfFunc = """
+import Foundation
+
+/// \(String.mockAnnotation)
+protocol NonSimpleFuncs {
+@discardableResult
+func returnSelf() -> Self
+}
+"""
+
+let returnSelfFuncMock = """
+
+import Foundation
+
+
+class NonSimpleFuncsMock: NonSimpleFuncs {
+    init() { }
+
+    var returnSelfCallCount = 0
+    var returnSelfHandler: (() -> (NonSimpleFuncsMock))?
+    func returnSelf() -> NonSimpleFuncsMock {
```

Comment:
The return type here should still be Self as the encloser type is not final.  We could make it final but I've seen cases where some mocks are subclassed for customization so will need to audit first.